yaksh/models.py

```diff
@@ -2062,6 +2063,39 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_diff_of_consecutive_submissions_for_all_code_question(self):
+        """
+            Return diff of consecutive submissions for all questions
+        """
+        q_a = {} # key : question, value : all submitted answers for this question
+        for answer in self.answers.all():
+            question = answer.question
+            if question in q_a:
+                q_a[question].append(answer)
+            else:
+                q_a[question] = [answer]
+
+        q_diff = {} # key : question, value : list of diff between consecutive submission for this question
+        for question in q_a:
+            if question.type=='code':
+                answers = q_a[question]
+                arr = [] #all answers for a question as list of lines
+                for ans in answers:
+                    split_ans_line_by_line = ans.answer.split("\n")
+                    split_ans_line_by_line = [ele+'\n' for ele in split_ans_line_by_line]
```

Comment:
You can use the `string.join` method.

Reply:
string.join method does not serve the purpose.. see example:
>>> s = "hello\nworld]\nwor\n"
>>> s.split('\n')
['hello', 'world]', 'wor', '']
>>> '\n'.join(s.split('\n'))
'hello\nworld]\nwor\n' This is not the required thing
>>> s = [ele+'\n' for ele in s.split('\n')]
>>> s
['hello\n', 'world]\n', 'wor\n', '\n'] This is the required one

Reply:
Since you are using list comprehension, you could have implemented this even more concisely:

```split_ans_line_by_line = [ele + '\n' for ele in ans.answer.split('\n') if ele]```